packages/js-config-webpack-plugin/src/JsConfigWebpackPlugin.js

```diff
@@ -33,13 +33,32 @@ class JsConfigWebpackPlugin {
 	 * @returns {string}
 	 */
 	resolveBabelConfigFilePath(contextPath, environmentName) {
+		/**
+		 * Return the value, unless its an interator, in which case get the last value.
+		 * Works around differences in return values of @babel/core functions between versions.
+		 * @template T
+		 * @param {Generator<T> | T} valueOrIterator
+		 * @returns {T}
+		 */
+		// prettier-ignore
```

Comment:
Why do we need a prettier ignore here?

Reply:
I have to add the following type assertion to keep `tsc` happy:

```
/** @type {Generator<T>} */ (valueOrIterator).next()
```

Prettier *thinks* it can improve this, and formats it without the brackets:

```
/** @type {Generator<T>} */ valueOrIterator.next()
```

However this breaks the type assertion as it now thinks:
- I am accessing the `next()` method of something that might not be a generator.
- I am casting the return value of `.next()` to be a generator instead,

Happy to implement an alternative suggestion 🙂